app/services/participants/query.rb

```diff
@@ -0,0 +1,74 @@
+module Participants
+  class Query
+    attr_reader :scope, :lead_provider
+
+    def initialize(lead_provider:)
+      @lead_provider = lead_provider
+      @scope = all_participants
+    end
+
+    def participants
+      scope.order(created_at: :asc)
+    end
+
+    def participant(id: nil)
+      scope.where(id:)
+    end
+
+  private
+
+    def all_participants
+      TrainingPeriod.from("(#{ect_participants.to_sql} UNION #{mentor_participants.to_sql}) as training_periods")
```

Comment:
similar to the questions above, I think we should be querying by teacher as thats the same as User in ecf1? happy to chat